internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -547,6 +630,104 @@ func (r *BMACReconciler) findBMH(ctx context.Context, agent *aiv1beta1.Agent) (*
+	bmhSpoke, mutateFn := r.newSpokeBMH(bmh, machine, agent)
+	if result, err := controllerutil.CreateOrUpdate(ctx, spokeClient, bmhSpoke, mutateFn); err != nil {
+		return nil, err
+	} else if result != controllerutil.OperationResultNone {
+		r.Log.Info("Spoke BareMetalHost created")
+	}
+	return bmhSpoke, nil
+}
+
+func (r *BMACReconciler) ensureSpokeMachine(ctx context.Context, spokeClient client.Client, bmh *bmh_v1alpha1.BareMetalHost, clusterDeployment *hivev1.ClusterDeployment) (*machinev1beta1.Machine, error) {
+	machineSpoke, mutateFn := r.newSpokeMachine(bmh, clusterDeployment)
+	if result, err := controllerutil.CreateOrUpdate(ctx, spokeClient, machineSpoke, mutateFn); err != nil {
+		return nil, err
+	} else if result != controllerutil.OperationResultNone {
+		r.Log.Info("Spoke Machine created")
+	}
+	return machineSpoke, nil
+}
+
+func (r *BMACReconciler) ensureSpokeBMHSecret(ctx context.Context, spokeClient client.Client, bmh *bmh_v1alpha1.BareMetalHost) (*corev1.Secret, error) {
+	secretName := bmh.Spec.BMC.CredentialsName
+	secret := &corev1.Secret{}
+	err := r.Get(ctx, types.NamespacedName{Namespace: bmh.Namespace, Name: secretName}, secret)
+	if err != nil {
+		return secret, err
+	}
+	secretSpoke, mutateFn := r.newSpokeBMHSecret(secret, bmh)
+	if result, err := controllerutil.CreateOrUpdate(ctx, spokeClient, secretSpoke, mutateFn); err != nil {
+		return nil, err
+	} else if result != controllerutil.OperationResultNone {
+		r.Log.Info("Spoke BareMetalHost Secret created")
+	}
+	return secretSpoke, nil
+}
+
+func (r *BMACReconciler) newSpokeBMH(bmh *bmh_v1alpha1.BareMetalHost, machine *machinev1beta1.Machine, agent *aiv1beta1.Agent) (*bmh_v1alpha1.BareMetalHost, controllerutil.MutateFn) {
+	bmhSpoke := &bmh_v1alpha1.BareMetalHost{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      bmh.Name,
+			Namespace: bmh.Namespace,
+		},
+	}
+	mutateFn := func() error {
+		bmhSpoke.Spec = bmh.Spec
+		// The host is created by the baremetal operator on hub cluster. So BMH on the spoke cluster needs
+		// to be set to externally provisioned
+		bmhSpoke.Spec.ExternallyProvisioned = true
+		// If the Image field is filled in, ExternallyProvisioned is ignored. So remove the Image field from spec
+		bmhSpoke.Spec.Image = &bmh_v1alpha1.Image{}
+		bmhSpoke.Spec.ConsumerRef.Name = machine.Name
+		bmhSpoke.Spec.ConsumerRef.Namespace = machine.Namespace
+		// copy annotations. hardwaredetails annotations is needed for automatic csr approval
+		bmhSpoke.ObjectMeta.Annotations = bmh.ObjectMeta.Annotations
+		// HardwareDetails annotation needs a special case. The annotation gets removed once it's consumed by the baremetal operator
+		// and data is copied to the bmh status. So we are reconciling the annotation from the agent status inventory.
+		// If HardwareDetails annotation is already copied from hub bmh.annotation above, this won't overwrite it.
+		r.reconcileAgentInventory(bmhSpoke, agent)
```

Comment:
Should the result from this function be checked? It's possible that an error occurs during this reconcile call. 

This could call `.Result()` and get the error from there.

Reply:
Done. Working on separate PR with unit tests.